File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        ArrayList<Integer> col=new ArrayList<>();
        ArrayList<Integer> row=new ArrayList<>();
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[i].length;j++){
                if(matrix[i][j]==0){
                    col.add(j);
                    row.add(i);
                }
            }
        }
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[i].length;j++){
                if(row.contains(i)){
                    Arrays.fill(matrix[i],0);
                }
                if(col.contains(j)){
                    matrix[i][j]=0;
                }
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        cln=set()
        row=set()
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if matrix[i][j]==0:
                    row.add(i)
                    cln.add(j)
        print(row)
        print(cln)
        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                if i in row or j in cln:
                    matrix[i][j]=0
        return matrix